modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -2556,3 +2560,34 @@ def _is_trivial_index(cls, index):
             and index.min() == 0
             and index.max() == len(index) - 1
         )
+
+
+class LazyProxyCategoricalDtype(pandas.CategoricalDtype):
+    """
+    Proxy class for lazily retrieving categorical dtypes from arrow tables.
+
+    Parameters
+    ----------
+    table : pyarrow.Table
+        Source table.
+    index : int
+        Column index.
+    """
+
+    def __init__(self, table: pyarrow.Table, index: int):
+        self._table = table
+        self._index = index
+        self._dtype = None
+
+    @property
+    def categories(self) -> Index:  # noqa: GL08
+        return self._delegate().categories
+
+    @property
+    def ordered(self):  # noqa: GL08
+        return self._delegate().ordered
+
+    def _delegate(self) -> pandas.CategoricalDtype:  # noqa: GL08
+        if self._dtype is None:
+            self._dtype = self._table.column(self._index).to_pandas().dtypes
+        return self._dtype
```

Comment:
`pandas.CategoricalDtype` seems to store actual categorical values inside `._categories` field [[link]](https://github.com/pandas-dev/pandas/blob/8dab54d6573f7186ff0c3b6364d5e4dd635ff3e7/pandas/core/dtypes/dtypes.py#L334-L342), can we just override accesing this specific field so all of the other methods would work automatically?

Reply:
can we also use an approach proposed in [the comment](https://github.com/modin-project/modin/pull/5271#issuecomment-1334456207) and extract categorical values from a pyarrow dictionary and not the whole column? I believe that would be much more efficient

Reply:
> can we also use an approach proposed in [the comment](https://github.com/modin-project/modin/pull/5271#issuecomment-1334456207) and extract categorical values from a pyarrow dictionary and not the whole column?

Do you mean self._table.column(self._index).chunks[0].dictionary.to_pandas().dtypes ?

Reply:
yes, but of course with categories union in case of multi-chunking

Reply:
type(self._table.column(self._index).chunks[0].dictionary.to_pandas().dtypes)
returns <class 'numpy.dtype[object_]'>

Reply:
yes, but wrapping the mentioned into `pandas.Categorical` will give us what we want:
```
pandas.Categorical(self._table.column(self.index).chunks[0].dictionary.to_pandas())
```

Reply:
Not actually the same:

> type(pandas.Categorical(self._table.column(self._index).chunks[0].dictionary.to_pandas().dtypes))
<class 'pandas.core.arrays.categorical.Categorical'>

> type(self._table.column(self._index).to_pandas().dtypes)
<class 'pandas.core.dtypes.dtypes.CategoricalDtype'>

> repr(pandas.Categorical(self._table.column(self._index).chunks[0].dictionary.to_pandas().dtypes))
'[object]
Categories (1, object): [object]'

> repr(self._table.column(self._index).to_pandas().dtypes)
"CategoricalDtype(categories=['first', 'second', 'third'], ordered=False)"

Reply:
okay, but since `pandas.CategoricalDtype._categories` field has `pandas.Index` type then all we need to do is just assing a `pandas.Index` containing values for categories:
```python
self._categories = pandas.Index(at.column(0).chunks[0].dictionary.to_pandas())
``` 
I assume this will be the same then

Reply:
> Not actually the same:

Okay, I was wrong previously, we should wrap the `.dictionary.to_pandas()` with a pandas.CategoricalDtype not with pandas.Categorical:
```python
>>> type(pandas.CategoricalDtype(at.column(0).chunks[0].dictionary.to_pandas()))
<class 'pandas.core.dtypes.dtypes.CategoricalDtype'>
>>> type(at.column(0).to_pandas().dtype)
<class 'pandas.core.dtypes.dtypes.CategoricalDtype'>
>>> pandas.CategoricalDtype(at.column(0).chunks[0].dictionary.to_pandas())
CategoricalDtype(categories=['v1', 'v2'], ordered=False)
>>> at.column(0).to_pandas().dtype
CategoricalDtype(categories=['v1', 'v2'], ordered=False)
```
this way they should be 100% equal

Reply:
Right, pandas.CategoricalDtype(self._table.column(self._index).chunks[0].dictionary.to_pandas()) is the same